plugins/inputs/http/http.go

```diff
@@ -0,0 +1,165 @@
+package http
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers"
+)
+
+type HTTP struct {
+	URLs []string `toml:"urls"`
+
+	// HTTP Basic Auth Credentials
+	Username string
+	Password string
+
+	// Path to CA file
+	SSLCA string `toml:"ssl_ca"`
+	// Path to host cert file
+	SSLCert string `toml:"ssl_cert"`
+	// Path to cert key file
+	SSLKey string `toml:"ssl_key"`
+	// Use SSL but skip chain & host verification
+	InsecureSkipVerify bool
+
+	Timeout internal.Duration
+
+	client *http.Client
+
+	// The parser will automatically be set by Telegraf core code because
+	// this plugin implements the ParserInput interface (i.e. the SetParser method)
+	parser parsers.Parser
+}
+
+var sampleConfig = `
+  ## One or more URLs from which to read formatted metrics
+  urls = [
+    "http://localhost:2015/simple.json"
```

Comment:
Because of my comment about `data_format` we should set this to something more format neutral such as `http://localhost/metrics`